.changes/next-release/enhancement-dependency-35338.json

```diff
@@ -0,0 +1,5 @@
+{
+  "type": "enhancement",
+  "category": "dependency",
+  "description": "Update backend dependency on PyInstaller to version 5.7.0"
```

Comment:
For this description, let's just make it `Add Python 3.11 support for source distribution`